### PR TITLE
Create script for importing service provider data into database

### PR DESCRIPTION
**Pull request by silastsui**
This script creates the `service_provider` table and imports data from a csv into the table. The data to be imported is assumed to be well-formed.

### CODE REVIEW

**Comment by silastsui** [comment]
I think we should make our table names consistent. I noticed in `setup_db.py` that `census_division` is in the singular form - thoughts on changing it? 

Also, it's probably worth talking about what fields should be not null in `service_providers`.

File: service_provider_data.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+2017        # good
+2016-2017   # bad
+2016/2017   # bad
+?           # bad
+```
+
+### Budget
+The budget should be the total expenses of the organization for the year. There should be no whitespace between numbers.
+
+TODO: verify that we want to use expenses as the operating budget
```
Comment (suzyng83209) [inline service_provider_data.md]: Yes

File: service_provider_data.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+```
+102248      # good
+102,248     # bad
+102 248     # bad
+```
+
+### Client Total, Staff Total
+These counts should be integers. Comments about the totals should be put in the `notes` column.
+
+TODO: how do we count clients served if the org provides multiple counts for multiple services
```
Comment (suzyng83209) [inline service_provider_data.md]: Sum up (don't mind duplication)

File: service_provider_data.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+TODO: how do we distinguish between youth clients and general clients
```
Comment (suzyng83209) [inline service_provider_data.md]: for now, save general client # and extrapolate from age demographic information

File: scripts/load_data_into_db.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import getpass
+import pandas as pd
+import psycopg2
+import sys
+
+def create_table(hostname, dbname, user):
```
Comment (Ruxuan) [inline scripts/load_data_into_db.py]: This is really similar to the code inside of setup_db.py. Can we move this to the `create_tables` function in setup_db.py?

File: scripts/load_data_into_db.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    commands = [
+            """
+            DROP TABLE IF EXISTS service_providers
+            """,
+            """
+            CREATE TABLE service_providers (
+                id INTEGER PRIMARY KEY,
+                census_division_id INTEGER,
```
Comment (Ruxuan) [inline scripts/load_data_into_db.py]: Mark `census_division_id` as a FOREIGN KEY similar to how the demographics tables does it.

File: scripts/load_data_into_db.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        if con:
+            con.rollback()
+        print("Error {}".format(e))
+        sys.exit(1)
+
+    if con:
+        con.close()
+
+def clean_service_provider_data():
+    filename = "service_provider_data.csv"
```
Comment (Ruxuan) [inline scripts/load_data_into_db.py]: Add filename to `constants.py`

File: scripts/load_data_into_db.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+
+    int_columns = ['report_year', 'expenses', 'client_total', 'staff_total']
+    str_columns = ['report_link', 'address', 'notes']
+    data[int_columns] = data[int_columns].fillna(0)
+    data[str_columns] = data[str_columns].fillna('')
+
+    return data
+
+def load_service_providers(hostname, dbname, user):
```
Comment (Ruxuan) [inline scripts/load_data_into_db.py]: I think you could put all these functions in `setup_db.py`. Every time we want to update service provider data or demographic data, we'll just update the other as well.

File: scripts/setup_db.py [hunk 8]
```diff
@@ -26,6 +31,24 @@ def create_tables(hostname, dbname, user):
...
+                id INTEGER PRIMARY KEY,
+                census_division_id INTEGER REFERENCES census_division (id),
+                name TEXT NOT NULL,
+                website TEXT,
+                report_year INTEGER,
+                report_link TEXT,
+                expenses INTEGER,
+                client_total INTEGER,
+                staff_total INTEGER,
+                address TEXT,
```
Comment (Ruxuan) [inline scripts/setup_db.py]: What's our plan for satellite addresses?

Comment (silastsui) [reply scripts/setup_db.py]: i think the addresses table should have a foreign key for `service_provider_id`. seems to be the best way to handle the one-to-many relationship.

thoughts?

**Pull request merged**